sieve: init at 0.6.1-unstable-2025-01-06

fugidev opened the pull request:
## Description of changes

Adds [Sieve](https://github.com/thsmi/sieve), an Electron-based sieve script editor, for Linux and Darwin.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fugidev committed:
sieve: init at 0.6.1-unstable-2025-01-06

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2099

eljamm commented on pkgs/by-name/si/sieve/package.nix:
```diff
@@ -0,0 +1,109 @@
+
+  src = fetchFromGitHub {
+    owner = "thsmi";
+    repo = pname;
```
It's better to use a string literal here (see https://github.com/NixOS/nixpkgs/issues/277994)

```suggestion
    repo = "sieve";
```

eljamm commented on pkgs/by-name/si/sieve/package.nix:
```diff
@@ -0,0 +1,109 @@
+  ];
+
+  meta = {
+    description = "A sieve script editor";
```
According to [the docs](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes), `meta.description` should not start with an indefinite article.

```suggestion
    description = "Sieve script editor";
```

eljamm commented on pkgs/by-name/si/sieve/package.nix:
```diff
@@ -0,0 +1,109 @@
+      cp -r $build_dir $out/share/sieve
+
+      # copy icon
+      icon_dir=$out/share/icons/hicolor/64x64/apps
+      mkdir -p $icon_dir
+      cp $build_dir/libs/icons/linux.png $icon_dir/sieve.png
```
The `icon_dir` variable seems redundant as it's only used once. We can also use `install` to create the destination and copy the file in one line:

```suggestion
      install -D $build_dir/libs/icons/linux.png $out/share/icons/hicolor/64x64/apps/sieve.png
```

fugidev commented on pkgs/by-name/si/sieve/package.nix:
```diff
@@ -0,0 +1,109 @@
+    license = lib.licenses.agpl3Only;
+    mainProgram = "sieve";
+    maintainers = with lib.maintainers; [ fugi ];
+    platforms = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
```
It should run on any architecture that electron supports, which doesn't include all of lib.platforms.darwin – maybe doing `platforms = electron.meta.platforms;` would be best?

eljamm replied:
That would indeed be better, thanks!

fugidev commented:
Thanks for the review @eljamm! I've applied your requested changes, updated to the latest upstream commit and rebased onto master. I'll squash after testing on aarch64-linux and aarch64-darwin.

eljamm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 331070`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sieve</li>
  </ul>
</details>

eljamm reviewed (approved):
I haven't tested the program as I don't know how to use it, but I trust that you did, so this looks good to me. Thanks a lot for your contribution!

h7x4 commented:
It seems this got added in #337931. It looks like there's quite a lot of stuff here that could be added to the existing package if you'd like. It might be better to open a new PR.

fugidev commented:
Thanks for letting me know. I'll be honest, it's quite disheartening to see a duplicate PR get merged so quickly, with nobody having checked if someone else had already made the effort to create one. :/
